src/libraries/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -453,6 +453,7 @@ public sealed partial class JsonSerializerOptions
     {
         public JsonSerializerOptions() { }
         public bool AllowTrailingCommas { get { throw null; } set { } }
+        public bool AllowPrivateProperties { get { throw null; } set { } }
```

Comment:
Since this is an API change, it would need to go through API review. Also, I don't think we want to support non-public properties within the JsonSerializer. I haven't seen evidence that a lot of folks want this feature.

Reply:
An issue with supporting non-public is that it won't work with code-gen \ AOT scenarios where we can't use reflection.